vtex/reliability-research
# Add batcher

#### What is the purpose of this pull request?

This PR is part of the changes discussed in #111
The `batcher.py` will be a new component for processing the load data, more specifically the documents for POST requests.
This script will be used to group our load documents in batches of arbitrary size, depending on your experiment's needs. 
Its implementation is as straightforward as it gets: read the documents and for each <`batch_size`> documents group them in a single file
Notice that the script will add `<add>` tags around the document's content to give the `replayer` an easier time creating the POST requests.


#### What problem is this solving?

- Improves the replayer's debugability, as you can trace an error to its corresponding document batch.
- Allows better control of the batch size for each experiment, necessary in future experiments  

#### How should this be manually tested?
You'll need a file with documents in `XML`, let's name it `docs`.
Then, run:
`python3 batcher.py --docs=./docs --size=50 --dir=./test`

This should create a folder `test` in the current directory and, inside it, batch files with 50 documents each.

#### Types of changes
- New feature (non-breaking change which adds functionality)
